### PR TITLE
core: fix when db empty on startup overrides for TTD and gray glacier do not work

### PR DESCRIPTION
**Pull request by leolara**
Fixes https://github.com/ethereum/go-ethereum/issues/25133

My first go-ethereum contribution, but I done some on prysm: https://github.com/prysmaticlabs/prysm/pulls?q=+is%3Apr+author%3Aleolara+

Please, let me know how to improve this PR.

### CODE REVIEW

File: core/genesis.go [hunk 1]
```diff
@@ -250,6 +250,7 @@ func SetupGenesisBlockWithOverride(db ethdb.Database, genesis *Genesis, override
 		if err != nil {
 			return genesis.Config, common.Hash{}, err
 		}
+		performConfigOverride(genesis.Config, overrideGrayGlacier, overrideTerminalTotalDifficulty)
```
Comment (leolara) [inline core/genesis.go:253]: perhaps we should clone genesis.Config before overriding it?

**Comment by karalabe** [comment]
Thanks, but there's already a PR open to tackle this https://github.com/ethereum/go-ethereum/pull/25136

**Comment by leolara** [comment]
@karalabe ok, no worries.

Is there a specific discord channel to discuss geth contributions, so I can talk about what I am doing before doing it?

File: core/genesis.go (hunk 1 above)
Comment (Danyalkasiri) [reply core/genesis.go:253]: 0x62044d2cab405388891ee6d53747817f34c0f00341cde548c0ce9834e9718f27

Comment (Danyalkasiri) [reply core/genesis.go:253]: 0x62044d2cab405388891ee6d53747817f34c0f00341cde548c0ce9834e9718f27

File: core/genesis.go [hunk 2]
```diff
@@ -322,6 +313,15 @@ func SetupGenesisBlockWithOverride(db ethdb.Database, genesis *Genesis, override
 	return newcfg, stored, nil
 }
 
+func performConfigOverride(cfg *params.ChainConfig, overrideGrayGlacier *big.Int, overrideTerminalTotalDifficulty *big.Int) {
```
[the remark this answers is not in the document]
Comment (Danyalkasiri) [reply core/genesis.go:316]: 0x62044d2cab405388891ee6d53747817f34c0f00341cde548c0ce9834e9718f27

**Pull request closed without merging**